algorithms/linked_list.py

```diff
@@ -1,57 +1,69 @@
-class Node():
-	def __init__(self, value, next):
-		self.value=value
-		self.next=next
+class Node:
+	def __init__(self, value, next = None):
+		self.value = value
+		self.next = next
 
-class LinkedList():
-	def __init__(self, head, tail):
-		self.head=head
-		self.tail=tail
+	def __repr__(self):
+		return "Node(value={}, next={})".format(self.value, self.next)
 
-	def add(self, node):
-		self.tail.next=node
-		self.tail=node
+class LinkedList:
+	def __init__(self, head):
+		self.head = head
+		self.tail = self.set_tail()
+
+	def __repr__(self):
+		return "LinkedList(head={}, tail={})".format(self.head, self.tail)
+
+	def set_tail(self):
```

Comment:
I think you actually don't need this method :) . Per this implementation, when you initialize the list up above, self.tail = head